core/common/src/PlanarClipMask.ts

```diff
@@ -0,0 +1,149 @@
+  public readonly modelIds?: CompressedId64Set;
+  /** The SubCategory ids if mode is [[PlanarClipMaskMode.IncludeSubCategorie]] or element Ids if mode is [[PlanarClipMaskMode.IncludeElements] or [[PlanarClipMaskMode.ExcludeElements]] */
+  public readonly subCategoryOrElementIds?: CompressedId64Set;
+  /** Priority value if mode is [[PlanarClipMaskMode.Priority]]. */
+  public readonly priority?: number;
+  /** A value between 0 and 1 indicating am override for mask transparency - A transparency of 0 (the default) indicates complete masking.  1 is completely transparent (no masking).  If no transparency is defined then the transparency of the mask elements are used */
+  public readonly transparency?: number;
+
+  /** Create a new [[PlanarClipMaskSettings]] object from a JSON object */
+  public static fromJSON(json?: PlanarClipMaskProps): PlanarClipMaskSettings {
+    if (!json)
+      return this.defaults;
+
+    return new PlanarClipMaskSettings(json.mode, json.transparency, json.modelIds, json.subCategoryOrElementIds, json.priority);
+  }
+  /** Create a [[PlanarClipMaskAettings]] object */
+  public static create(mode: PlanarClipMaskMode, modelIds?: Id64Set, subCategoryOrElementIds?: Id64Set, transparency?: number): PlanarClipMaskSettings | undefined {
+    switch (mode) {
+      case PlanarClipMaskMode.None:
+      case PlanarClipMaskMode.Priority:
+        return new PlanarClipMaskSettings(mode);
+
+      case PlanarClipMaskMode.Models:
+        return modelIds === undefined ? undefined : new PlanarClipMaskSettings(mode, transparency, CompressedId64Set.compressSet(modelIds));
+
+      case PlanarClipMaskMode.IncludeSubCategories:
+      case PlanarClipMaskMode.IncludeElements:
+      case PlanarClipMaskMode.ExcludeElements:
+        return subCategoryOrElementIds === undefined ? undefined : new PlanarClipMaskSettings(mode, transparency, modelIds ? CompressedId64Set.compressSet(modelIds) : undefined, CompressedId64Set.compressSet(subCategoryOrElementIds));
+
+      default:
+        return undefined;
+    }
+  }
+
+  /** Create [[planarClipMaskSettings]] object by priority
+   * @see [[PlanarClipMaskPriority]]
+   */
+  public static createByPriority(priority: number, transparency?: number) {
+    return new PlanarClipMaskSettings(PlanarClipMaskMode.Priority, transparency, undefined, undefined, priority);
+  }
+
+  /** Create JSON object representing this [[PlanarClipMaskSettings]] */
+  public toJSON(): PlanarClipMaskProps {
+    return { mode: this.mode, modelIds: this.modelIds, subCategoryOrElementIds: this.subCategoryOrElementIds, priority: this.priority, transparency: this.transparency  };
+  }
+
+  /**  */
+  public get isValid(): boolean { return this.mode !== PlanarClipMaskMode.None; }
+
+  public equals(other: PlanarClipMaskSettings): boolean {
+    return this.mode === other.mode &&
+      compareNumbersOrUndefined(this.priority, other.priority) === 0 &&
```

Comment:
Why not use `===` instead of `compareXxxOrUndefined`?

Reply:
Good question...  why do we even have compareNumbersOrUndefined?

Reply:
For **ordered** comparisons.

Reply:
Fixed.